plugins/destination/postgresql/client/migrate.go

```diff
@@ -167,6 +170,50 @@ func (c *Client) dropTable(ctx context.Context, tableName string) error {
 	return nil
 }
 
+func (c *Client) migrateToCQID(ctx context.Context, tableName string, _ schema.Column) error {
+	// Steps:
+	// acquire connection
+	conn, err := c.conn.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to acquire connection: %w", err)
+	}
+	defer conn.Release()
+	// start transaction
+	tx, err := conn.BeginTx(ctx, pgx.TxOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+
+	sanitizedTableName := pgx.Identifier{tableName}.Sanitize()
+	sanitizedPKName := pgx.Identifier{getPKName(&schema.Table{Name: tableName})}.Sanitize()
+
+	// Drop existing primary key
+	_, err = tx.Exec(ctx, "alter table "+sanitizedTableName+" drop constraint "+sanitizedPKName)
+	if err != nil {
+		c.logger.Error().Err(err).Str("table", tableName).Msg("Failed to drop primary key")
+		if rollbackErr := tx.Rollback(ctx); rollbackErr != nil {
+			return fmt.Errorf("failed to rollback transaction: %w", rollbackErr)
+		}
+		return err
+	}
+
+	// Create new Primary Key with CQID
+	_, err = tx.Exec(ctx, "ALTER TABLE "+sanitizedTableName+" ADD CONSTRAINT "+sanitizedPKName+" Primary Key ("+pgx.Identifier{schema.CqIDColumn.Name}.Sanitize()+")")
+	if err != nil {
+		c.logger.Error().Err(err).Str("table", tableName).Msg("Failed to create new primary key on _cq_id")
+		if rollbackErr := tx.Rollback(ctx); rollbackErr != nil {
+			return fmt.Errorf("failed to rollback transaction: %w", rollbackErr)
+		}
+		return err
+	}
+	// Commit transaction if no error
+	err = tx.Commit(ctx)
```

Comment:
this can be rolled into a commit/rollback pattern like this:
```go
func A() (err error) {
...
	tx, err := conn.BeginTx(ctx, pgx.TxOptions{})
	if err != nil {
		return err
	}
	defer func() {
		if err == nil {
			if err = tx.Commit(ctx); err != nil {
				err = fmt.Errorf("failed to commit transaction: %w", err)
			}
		}

		// this will also work for tx commit errors
		if err != nil {
			if rollbackErr := tx.Rollback(ctx); rollbackErr != nil {
				// we want to preserve the original err still, so only log
				c.logger.Err(rollbackErr).Msg("failed to rollback transaction")
			}
		}
	}()
	...
}
```